packages/navigator/src/browser/navigator-tab-bar-decorator.ts

```diff
@@ -0,0 +1,64 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { TabBarDecorator } from '@theia/core/lib/browser/shell/tab-bar-decorator';
+import { EXPLORER_VIEW_CONTAINER_ID } from './navigator-widget';
+import { Title, Widget, FrontendApplicationContribution, FrontendApplication, Navigatable, Saveable, ApplicationShell } from '@theia/core/lib/browser';
+import { WidgetDecoration } from '@theia/core/lib/browser/widget-decoration';
+import { DisposableCollection, Disposable } from '@theia/core/lib/common/disposable';
+
+@injectable()
+export class NavigatorTabBarDecorator implements TabBarDecorator, FrontendApplicationContribution {
+    readonly id = 'theia-navigator-tabbar-decorator';
+    protected shell: ApplicationShell;
+
+    protected readonly emitter = new Emitter<void>();
+    private readonly toDispose = new DisposableCollection();
+    private readonly toDisposeOnDirtyChanged = new Map<Widget, Disposable>();
+
+    onStart(app: FrontendApplication): void {
+        this.shell = app.shell;
+        this.toDispose.push(this.shell.onDidAddWidget(w => {
```

Comment:
we don't short variable names: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#whole-words-names